### PR TITLE
rls: use channel creds to create resolvingOobChannel

### PR DESCRIPTION
**Pull request by dapengzhang0**

### CODE REVIEW

File: rls/src/main/java/io/grpc/rls/CachingRlsLbClient.java [hunk 1]
```diff
@@ -139,8 +139,9 @@ private CachingRlsLbClient(Builder builder) {
             timeProvider);
     RlsRequestFactory requestFactory = new RlsRequestFactory(lbPolicyConfig.getRouteLookupConfig());
     rlsPicker = new RlsPicker(requestFactory);
-    ManagedChannelBuilder<?> rlsChannelBuilder =
-        helper.createResolvingOobChannelBuilder(rlsConfig.getLookupService());
+    ManagedChannelBuilder<?> rlsChannelBuilder = helper.createResolvingOobChannelBuilder(
+        rlsConfig.getLookupService(), helper.getUnsafeChannelCredentials());
```
Comment (ejona86) [inline rls/src/main/java/io/grpc/rls/CachingRlsLbClient.java:143]: Add a comment describing why it is safe to use the unsafe credentias.

File: rls/src/main/java/io/grpc/rls/CachingRlsLbClient.java [hunk 2]
```diff
@@ -139,8 +139,9 @@ private CachingRlsLbClient(Builder builder) {
...
+    rlsChannelBuilder.overrideAuthority(helper.getAuthority());
```
Comment (ejona86) [inline rls/src/main/java/io/grpc/rls/CachingRlsLbClient.java:144]: Has overrideAuthority been enhanced to make this safe?

Comment (dapengzhang0) [reply rls/src/main/java/io/grpc/rls/CachingRlsLbClient.java:144]: Just send out #7854

File: rls/src/main/java/io/grpc/rls/CachingRlsLbClient.java (hunk 1 above)
Comment (dapengzhang0) [reply rls/src/main/java/io/grpc/rls/CachingRlsLbClient.java:147]: Done.

**Review by ejona86 — APPROVED** [review]
This looks fine, but will need to wait until #7854 is merged.

**Pull request merged**